src/runtime_src/ert/scheduler/scheduler.cpp

```diff
@@ -1155,7 +1155,7 @@ scheduler_loop()
           continue; // CU is not used
 
         auto cuvalue = read_reg(cu_idx_to_addr(cuidx));
-        if (!(cuvalue & AP_DONE))
+        if (!(cuvalue & AP_DONE|AP_IDLE))
```

Comment:
Please check https://github.com/Xilinx/XRT/pull/1372.   
I am sorry that the comment on that pull request is not complete enough to explain the change that was made, but I am pretty sure you shouldn't be reverting it.

Reply:
The code should be fixed and "()" should be added appropriately.

More importantly, we need to add some comments here to explain why this change is required. Basically, both ERT and KDS are babysitting data flow CUs. By the time when ERT gets to check the CU status, it might have finished the job by KDS and in an idle state. If ERT does not check AP_IDLE, it will keep coming back and check the status until it sees AP_DONE, which might not happen since the application has finished. This is not ideal and will trigger firewall trip if, at the time, an xclbin download is happening.